layer1/executor/task_handler_test.go

```diff
@@ -0,0 +1,472 @@
+package executor
+
+import (
+	"context"
+	"errors"
+	"github.com/alicenet/alicenet/bridge/bindings"
+	"github.com/alicenet/alicenet/consensus/objs"
+	"github.com/alicenet/alicenet/constants"
+	"github.com/alicenet/alicenet/crypto"
+	"github.com/alicenet/alicenet/layer1/executor/tasks/dkg"
+	"github.com/alicenet/alicenet/layer1/executor/tasks/dkg/state"
+	taskMocks "github.com/alicenet/alicenet/layer1/executor/tasks/mocks"
+	"github.com/alicenet/alicenet/layer1/executor/tasks/snapshots"
+	snapshotState "github.com/alicenet/alicenet/layer1/executor/tasks/snapshots/state"
+	"github.com/alicenet/alicenet/layer1/transaction"
+	"github.com/alicenet/alicenet/test/mocks"
+	"github.com/dgraph-io/badger/v2"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+	"math/big"
+	"testing"
+	"time"
+)
+
+func getTaskHandler(t *testing.T, doCleanup bool) (*Handler, *mocks.MockClient, *mocks.MockAllSmartContracts, *mocks.MockWatcher, accounts.Account) {
+	db := mocks.NewTestDB()
+	client := mocks.NewMockClient()
+	client.GetFinalizedHeightFunc.SetDefaultReturn(0, nil)
+	adminHandlers := mocks.NewMockAdminHandler()
+	adminHandlers.IsSynchronizedFunc.SetDefaultReturn(true)
+	txWatcher := mocks.NewMockWatcher()
+	contracts := mocks.NewMockAllSmartContracts()
+
+	account := accounts.Account{
+		Address: common.HexToAddress("546F99F244b7B58B855330AE0E2BC1b30b41302F"),
+		URL: accounts.URL{
+			Scheme: "http",
+			Path:   "",
+		},
+	}
+
+	taskHandler, err := NewTaskHandler(db, client, contracts, adminHandlers, txWatcher)
+	require.Nil(t, err)
+
+	if doCleanup {
+		t.Cleanup(func() {
+			taskHandler.Close()
+			db.DB().Close()
+		})
+	}
+
+	taskHandler.Start()
+	return taskHandler.(*Handler), client, contracts, txWatcher, account
+}
+
+// getTaskManagerCopy creates a copy of the manager from the DB without race
+// conditions.
+func getTaskManagerCopy(t *testing.T, manager *TaskManager) *TaskManager {
+	newManager := &TaskManager{Schedule: make(map[string]ManagerRequestInfo), Responses: make(map[string]ManagerResponseInfo), marshaller: getTaskRegistry(), database: manager.database}
+	<-time.After(10 * time.Millisecond)
+	err := newManager.loadState()
+	if err != nil {
+		require.Equal(t, badger.ErrKeyNotFound, err)
+	}
+	return newManager
+}
+
+// getScheduleLen returns the amount of tasks is in the TaskManager.Schedule
+func getScheduleLen(t *testing.T, manager *TaskManager) int {
+	newManager := getTaskManagerCopy(t, manager)
+	return len(newManager.Schedule)
+}
+
+func TestTasksHandlerAndManager_Schedule_NilTask(t *testing.T) {
+	handler, _, _, _, _ := getTaskHandler(t, true)
```

Comment:
Consider marking all these tests as `t.Parallel()` to speed execution.

Reply:
added to some of the tests